### DISCUSSION TITLE
Get socket address of connection

### DISCUSSION THREAD

**Discussion by tp971** [actix/actix-web#3006] [Ideas] [answered]
I would like to have a server that has multiple binds like this:
```rs
HttpServer::new(move || {
    ...
})
.bind(("0.0.0.0", 80))?
.bind(("127.0.0.1", 8080))?
```
With that, I want certain routes to only be accessible using the second bind `("127.0.0.1", 8080)`, however there seems to be no way to distinguish through which bind the connection has been made.

**Comment by RoDmitry** [comment]
I think you can have two `HttpServer::new()` in main, with different routes and bind. And then await them with [`join!`](https://rust-lang.github.io/async-book/06_multiple_futures/02_join.html) or `tokio::spawn`. Like:
```rust
let server1 = HttpServer::new(move || {
        ...
    })
    .bind("0.0.0.0:80")?
    .run();
let server2 = HttpServer::new(move || {
        ...
    })
    .bind("127.0.0.1:8080")?
    .run();
tokio::spawn(server1); // not sure if it will work in #[actix_web::main]
server2.await;
```

**Comment by robjtede** [comment, marked as answer]
```rust
req.app_config().local_addr()
```

See https://docs.rs/actix-web/latest/actix_web/struct.HttpRequest.html#method.app_config
